seacrowd/sea_datasets/vintext/vintext.py

```diff
@@ -0,0 +1,214 @@
+_GDRIVE_ID = "1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml"
+
+_SUPPORTED_TASKS = [Tasks.OPTICAL_CHARACTER_RECOGNITION]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class VintextDataset(datasets.GeneratorBasedBuilder):
+    """Vintext is a challenging scene text dataset for Vietnamese, where some characters are equivocal in the visual form due to accent symbols.
+    This dataset contains 2000 fully annotated images with 56,084 text instances. Each text instance is delineated by a quadrilateral bounding box and associated with the ground truth sequence of characters.
+    The dataset is randomly split into three subsets for training (1,200 images), validation (300 images), and testing (500 images)."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_imtext",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema="seacrowd_imtext",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "image_path": datasets.Value("string"),
+                    "annotations": datasets.Sequence(
+                        {
+                            "x1": datasets.Value("int32"),
+                            "y1": datasets.Value("int32"),
+                            "x2": datasets.Value("int32"),
+                            "y2": datasets.Value("int32"),
+                            "x3": datasets.Value("int32"),
+                            "y3": datasets.Value("int32"),
+                            "x4": datasets.Value("int32"),
+                            "y4": datasets.Value("int32"),
+                            "transcript": datasets.Value("string"),
+                        }
+                    ),
+                }
+            )
+
+        elif self.config.schema == "seacrowd_imtext":
+            features = schemas.image_text_features()
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        zip_filepath = os.path.join(os.path.dirname(__file__), "vietnamese_original.zip")
```

Comment:
Clever! I never thought that one can get HF's `.cache` folder like this. 😄
I guess you know already that there is a caveat on using cache folder though: once you edit the dataloader file (even a slight edit), the cache dir will be different, thus downloading the dataset again.

I personally store downloaded `gdown` data in a newly-created `data` directory in seacrowd root folder, though there are pros and cons.

Reply:
Thank you ahahaha 🙏.

My only concern is I am trying to minimize adding new folder to the SEACrowd repo that does not follow the template. But both is possible though!

The other consideration is, I think the hassle is only from the developer side (the slight change new cache problem). I don't think if one uses our dataset through `load_dataset` the cache will reset (ofc because there is no change to the dataloader file).

Again, please lmk if the `data` directory approach is cleaner! I would be happy to do it. Wdyt guys? @akhdanfadh @holylovenia @fhudi

Reply:
@patrickamadeus Either implementation should be fine as long as it works. Let's keep the current implementation since it has been done.